### PR TITLE
implement digamma,trigamma using polygamma

### PR DESCRIPTION
**Pull request by ranjithkumar007**
- [x] tests to be added

### CODE REVIEW

**Comment by isuruf** [comment]
This PR contains the erroneous commit, 855a295. You should remove it

**Comment by ranjithkumar007** [comment]
@isuruf can you restart this [job](https://travis-ci.org/symengine/symengine/jobs/197050677)  ?

**Comment by ranjithkumar007** [comment]
@isuruf  Please review and suggest improvements !!

File: symengine/functions.h [hunk 1]
```diff
@@ -1115,6 +1116,16 @@ class PolyGamma : public TwoArgFunction
 RCP<const Basic> polygamma(const RCP<const Basic> &n,
                            const RCP<const Basic> &x);
 
+inline RCP<const Basic> digamma(const RCP<const Basic> &x)
+{
+    return polygamma(zero, x);
```
Comment (isuruf) [inline symengine/functions.h]: Can you move the implementation to the `.cpp` file?

**Comment by ranjithkumar007** [comment]
updated @isuruf

**Comment by isuruf** [comment]
Can you add simple, but important tests like, `digamma(x) == polygamma(0, x)` and `trigamma(x) == polygamma(1, x)` and `digamma(x)->diff(x) == trigamma(x)` ?

**Comment by ranjithkumar007** [comment]
this PR is ready for a final review !!

**Comment by ranjithkumar007** [comment]
ping @isuruf

**Pull request merged**